nautobot/extras/management/__init__.py

```diff
@@ -118,67 +154,109 @@ def create_custom_statuses(
 
         for choice_kwargs in choices:
-            # The value of `color` may differ between other enums. We'll need to
-            # make sure they are normalized in `export_statuses_from_choiceset`
-            # when we go to add `status` field for other object types.
+            # Since statuses are customizable now, we need to gracefully handle the case where a status
+            # has had its name, slug, color and/or description changed from the defaults.
+            # First, try to find by slug
+            defaults = choice_kwargs.copy()
+            slug = defaults.pop("slug")
             try:
-                obj, created = Status.objects.get_or_create(**choice_kwargs)
-            # This will likely be a duplicate key violation due to a Status
-            # already existing (e.g. "active") albeit with a different `color`.
-            # Pop the color and try again.
+                # may fail if a status with a different slug has a name matching this one
+                obj, created = Status.objects.get_or_create(slug=slug, defaults=defaults)
             except IntegrityError:
-                choice_kwargs.pop("color")
-                obj, created = Status.objects.get_or_create(**choice_kwargs)
-            # If this subsequent .get_or_create fails, fail immediately.
-            except Exception as err:
-                raise SystemExit(
-                    f"Unexpected error while running data migration to populate" f"status for {model_path}: {err}"
-                )
+                # OK, what if we look up by name instead?
+                defaults = choice_kwargs.copy()
+                name = defaults.pop("name")
+                try:
+                    obj, created = Status.objects.get_or_create(name=name, defaults=defaults)
+                except IntegrityError as err:
+                    raise SystemExit(
+                        f"Unexpected error while running data migration to populate status for {model_path}: {err}"
+                    )
 
             # Make sure the content-type is associated.
             if content_type not in obj.content_types.all():
                 obj.content_types.add(content_type)
 
             if created and verbosity >= 2:
-                print(f"      Adding and linking status {obj.name}", flush=True)
+                print(f"      Adding and linking status {obj.name} ({obj.slug})", flush=True)
                 added_total += 1
             elif not created and verbosity >= 2:
-                print(f"      Linking to existing status {obj.name}", flush=True)
+                print(f"      Linking to existing status {obj.name} ({obj.slug})", flush=True)
                 linked_total += 1
 
     if verbosity >= 2:
         print(f"    Added {added_total}, linked {linked_total} status records")
+
+
+def clear_status_choices(
+    apps=global_apps,
+    schema_editor=None,
+    verbosity=2,
+    models=list(CHOICESET_MAP.keys()),
```

Comment:
Same here as the previous comment for `models=` pls
```suggestion
    models=None,
```